crates/sui-core/src/transaction_orchestrator.rs

```diff
@@ -259,6 +361,75 @@ where
             in_flight.dec();
         });
 
+        // Check if TransactionDriver should be used for submission
+        if self.td_percentage > 0 {
+            let random_value = rand::thread_rng().gen_range(1..=100);
+            if random_value <= self.td_percentage {
+                debug!("Using TransactionDriver for transaction {:?}", tx_digest);
+                if let Some(td) = &self.transaction_driver {
+                    // Add transaction to WAL log for TransactionDriver path
+                    if self
+                        .pending_tx_log
+                        .write_pending_transaction_maybe(&transaction)
```

Comment:
is my understanding correctly that during recovery we'll attempt to re-send the transactions via quorum driver right? I guess for now that's fine